### PR TITLE
Show that there are no locks

### LINKED ISSUE #1097: Show that there are no lock yet?
When a creator comes to the dashboard for the first time, we should show that no lock has been previously deployed. Right now it's a big and scary void ;)

image

cc @smombartz for the design/wording of this?

### PR DESCRIPTION
**Pull request by cnasc**
# Description
This PR fixes #1097. It also depends on #1218 (for changes to the dashboard stories). When there are no locks, a message displays indicating that the creator needs to make a lock. This message does not display when the lock form is active, but reappears if the form is canceled.

screen shot 2019-01-25 at 1 34 18 pm
screen shot 2019-01-25 at 1 34 27 pm

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

PS: [Read how to write the perfect pull request](https://blog.github.com/2015-01-21-how-to-write-the-perfect-pull-request/)

### CODE REVIEW

File: unlock-app/src/components/creator/CreatorLocks.js [hunk 1]
```diff
@@ -54,23 +54,36 @@ export class CreatorLocks extends React.Component {
...
+          lockFeed.map(lock => {
+            return <CreatorLock key={JSON.stringify(lock)} lock={lock} />
+          })}
+        {lockFeed.length === 0 && !showDashboardForm && (
+          <Container>
+            <Image src="/static/images/illustrations/lock.svg" />
+            <DimMessage>
+              <h1>Create a lock to get started</h1>
+              <p>
+                If you have already created some locks, they should appear here
```
Comment (julien51) [inline unlock-app/src/components/creator/CreatorLocks.js]: 👍

Comment (julien51) [reply unlock-app/src/components/creator/CreatorLocks.js]: I wonder if a cleaner approach would not be to create a dedicated component for this?

Comment (julien51) [reply unlock-app/src/components/creator/CreatorLocks.js]: or maybe use `DefaultError` ?

File: unlock-app/src/services/web3Service.js [hunk 2]
```diff
@@ -259,6 +259,9 @@ export default class Web3Service extends EventEmitter {
       })
       .once('receipt', receipt => {
         callback(null, { event: 'receipt', args: { receipt } })
+        this.emit('transaction.updated', transaction, {
```
Comment (julien51) [inline unlock-app/src/services/web3Service.js]: This probably belong in its own PR?

File: unlock-app/src/components/creator/CreatorLocks.js (hunk 1 above)
Comment (cnasc) [reply unlock-app/src/components/creator/CreatorLocks.js]: Sorry there's a bit of noise in here, this is based on the lock sorting branch, so there are commits from that in here. I wasn't sure what the best way to create a dependent PR was...

**Comment by cnasc** [comment]
> Hey, I think this PR does 2 different things:
> 
>     * display locks as sorted
> 
>     * display a placeholder when there are no locks.
> 
> 
> I think it would be cleaner to split it in 2 different PRs. This would make reviewing easier ;)

Sorry there's a bit of noise in here, this is based on the lock sorting branch, so there are commits from that in here. I wasn't sure what the best way to create a dependent PR was...
